addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/discovery/DSMRBridgeDiscoveryService.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal.discovery;
+
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.dsmr.DSMRBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPortIdentifier;
+
+/**
+ * This implements the discovery service for detecting new DSMR Meters
+ *
+ * @author M. Volaart
+ * @since 2.1.0
+ */
+public class DSMRBridgeDiscoveryService extends AbstractDiscoveryService implements DSMRBridgeDiscoveryListener {
+    private final Logger logger = LoggerFactory.getLogger(DSMRBridgeDiscoveryService.class);
+
+    /**
+     * Constructs a new DSMRBridgeDiscoveryService
+     */
+    public DSMRBridgeDiscoveryService() {
+        super(Collections.singleton(DSMRBindingConstants.THING_TYPE_DSMR_BRIDGE),
+                DSMRBindingConstants.DSMR_DISCOVERY_TIMEOUT_SECONDS, false);
+    }
+
+    /**
+     * Starts a new discovery scan.
+     *
+     * All available Serial Ports are scanned for P1 telegrams.
```

Comment:
Is this really a safe thing to do? Couldn't that have the weirdest impacts on some connected serial devices which are no meters?

Reply:
I think it can be done because there won't be sent any date to the particular serial port (only reading). If the ports to other devices are already in use we skip these ports. 
So we will try only reading on serial ports that are not claimed by any other software application.